vault/activity_log_test.go

```diff
@@ -1390,6 +1390,80 @@ func setupActivityRecordsInStorage(t *testing.T, base time.Time, includeEntities
 	return a, entityRecords, tokenRecords
 }
 
+//setupActivityRecordsInStorageRootNS creates entity and non-entity tokens only with root namespace
+func setupActivityRecordsInStorageRootNS(t *testing.T, base time.Time, includeEntities, includeTokens bool) (*ActivityLog, []*activity.EntityRecord, map[string]uint64) {
+	t.Helper()
+
+	core, _, _ := TestCoreUnsealed(t)
+	a := core.activityLog
+	monthsAgo := base.AddDate(0, -3, 0)
+
+	var entityRecords []*activity.EntityRecord
+	if includeEntities {
+		entityRecords = []*activity.EntityRecord{
+			{
+				EntityID:    "11111111-1111-1111-1111-111111111111",
+				NamespaceID: "root",
+				Timestamp:   time.Now().Unix(),
+			},
+			{
+				EntityID:    "22222222-2222-2222-2222-222222222222",
+				NamespaceID: "root",
+				Timestamp:   time.Now().Unix(),
+			},
+			{
+				EntityID:    "33333333-2222-2222-2222-222222222222",
+				NamespaceID: "root",
+				Timestamp:   time.Now().Unix(),
+			},
+		}
+
+		testEntities1 := &activity.EntityActivityLog{
+			Entities: entityRecords[:1],
+		}
+		entityData1, err := proto.Marshal(testEntities1)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		testEntities2 := &activity.EntityActivityLog{
+			Entities: entityRecords[1:2],
+		}
+		entityData2, err := proto.Marshal(testEntities2)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+		testEntities3 := &activity.EntityActivityLog{
+			Entities: entityRecords[2:],
+		}
+		entityData3, err := proto.Marshal(testEntities3)
+		if err != nil {
+			t.Fatalf(err.Error())
+		}
+
+		WriteToStorage(t, core, ActivityLogPrefix+"entity/"+fmt.Sprint(monthsAgo.Unix())+"/0", entityData1)
+		WriteToStorage(t, core, ActivityLogPrefix+"entity/"+fmt.Sprint(base.Unix())+"/0", entityData2)
+		WriteToStorage(t, core, ActivityLogPrefix+"entity/"+fmt.Sprint(base.Unix())+"/1", entityData3)
+	}
+
+	var tokenRecords map[string]uint64
+	if includeTokens {
```

Comment:
everything up to this point looks identical with `setupActivityRecordsInStorage` - it might be a good idea to break some of that out into a base/helper function, and then just add the namespace flavor on top for the namespace version of the function. this would result in more clear and maintainable code